storops/vnx/resource/nqm.py

```diff
@@ -0,0 +1,317 @@
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+from __future__ import unicode_literals
+
+from contextlib import contextmanager
+
+from storops.vnx.resource import VNXCliResourceList, VNXCliResource
+from storops.vnx.resource.lun import VNXLun, VNXLunList
+from storops.vnx.enums import VNXCtrlMethod
+from storops import exception as ex
+
+__author__ = "Peter Wang"
+
+
+def convert_lun(luns):
+    lun_ids = []
+    smp_names = []
+    for lun in luns:
+        if lun.is_snap_mount_point:
+            smp_names.append(lun.name)
+        else:
+            lun_ids.append(lun.lun_id)
+    return lun_ids, smp_names
+
+
+def normalize_lun(luns, cli):
+    if isinstance(luns, int):
+        return [VNXLun(lun_id=luns, cli=cli)]
+    elif isinstance(luns, VNXLun):
+        return [luns]
+    elif isinstance(luns, list) or isinstance(luns, VNXLunList):
+        return luns
+    elif luns is None:
+        return []
+    else:
+        raise ValueError('Invalid format for luns.')
+
+
+@contextmanager
+def stop_run_policy(policy):
+    VNXIOPolicy.stop_policy(cli=policy._cli)
+    try:
+        yield policy
+    except ex.StoropsException:
+        pass
+    policy.run_policy()
+
+
+class VNXIOClassList(VNXCliResourceList):
+    def __init__(self, cli=None, name=None):
+        super(VNXIOClassList, self).__init__()
+        self._cli = cli
+        self._name = name
+
+    @classmethod
+    def get_resource_class(cls):
+        return VNXIOClass
+
+    def _get_raw_resource(self):
+        return self._cli.get_ioclass(name=self._name, poll=self.poll)
+
+    def _filter(self, item):
+        """Filter the system default class"""
+        if 'Background Class' in item.name:
+            return False
+        return True
+
+
+class VNXIOClass(VNXCliResource):
+    def __init__(self, name=None, cli=None):
+        super(VNXIOClass, self).__init__()
+        self._cli = cli
+        self._name = name
+
+    def _get_raw_resource(self):
+        if self._cli is None:
+            raise ValueError('client is not available for this resource.')
+        return self._cli.get_ioclass(name=self._name, poll=self.poll)
+
+    @staticmethod
+    def get(cli, name=None):
+        if name:
+            return VNXIOClass(name=name, cli=cli)
+        return VNXIOClassList(cli=cli)
+
+    @property
+    def luns(self):
+        """Aggregator for ioclass_luns and ioclass_snapshots."""
+        lun_list, smp_list = [], []
+        if self.ioclass_luns:
+            lun_list = map(lambda l: VNXLun(lun_id=l.lun_id, name=l.name,
```

Comment:
keep this for now, created a enhancement for future: https://github.com/emc-openstack/storops/issues/101